python/openai/openai_frontend/engine/triton_engine.py

```diff
@@ -299,44 +394,170 @@ def _get_first_streaming_chat_response(
-            logprobs=None,
-            finish_reason=ChatCompletionFinishReason.stop if response.final else None,
-        )
-
-        chunk = self._get_streaming_chat_response_chunk(
-            choice, request_id, created, model
-        )
-        return chunk
-
     async def _streaming_chat_iterator(
         self,
         request_id: str,
+        backend: str,
         created: int,
-        model: str,
+        request: CreateChatCompletionRequest,
         role: str,
+        tool_call_parser: ToolCallParser,
         responses: AsyncIterable,
     ) -> AsyncIterator[str]:
+        model = request.model
+
+        if request.tool_choice and isinstance(
+            request.tool_choice.root, ChatCompletionNamedToolChoice
+        ):
+            tool_choice_function_name = request.tool_choice.root.function.name
+        else:
+            tool_choice_function_name = None
+
+        if (
+            request.tool_choice
+            and request.tool_choice.root == ChatCompletionToolChoiceOption1.required
+        ):
+            tool_choice_required_function_name = request.tools[0].function.name
+        else:
+            tool_choice_required_function_name = None
+
+        # Determine whether tools are in use with "auto" tool choice
+        tool_choice_auto = (
+            tool_call_parser
+            and not tool_choice_function_name
+            and self._should_stream_with_auto_tool_parsing(request)
+        )
+
+        previous_text = "" if tool_choice_auto else None
+
         chunk = self._get_first_streaming_chat_response(
             request_id, created, model, role
         )
         yield f"data: {chunk.model_dump_json(exclude_unset=True)}\n\n"
 
         async for response in responses:
-            chunk = self._get_nth_streaming_chat_response(
-                request_id, created, model, response
+            delta_text = _get_output(response)
+
+            response_delta: Optional[ChatCompletionStreamResponseDelta]
+            if tool_choice_function_name:
+                response_delta = ChatCompletionStreamResponseDelta(
+                    tool_calls=[
+                        ChatCompletionMessageToolCallChunk(
+                            index=0,
+                            function=Function2(
+                                name=tool_choice_function_name, arguments=delta_text
+                            ),
+                        )
+                    ]
+                )
+            elif tool_choice_required_function_name:
+                response_delta = ChatCompletionStreamResponseDelta(
+                    tool_calls=[
+                        ChatCompletionMessageToolCallChunk(
+                            index=0,
+                            function=Function2(
+                                name=tool_choice_required_function_name,
+                                arguments=delta_text,
+                            ),
+                        )
+                    ]
+                )
```

Comment:
These first if two blocks have the same result right? Can it be simplified to something like this? (may not be 100% correct):
```python
tool_fn_name = tool_choice_function_name or tool_choice_required_function_name

if tool_fn_name:
  <call>
elif tool_choice_auto:
  ...
else:
  ...
```